### ISSUE TITLE
Wrong processing if basePath is present in swagger definition

### ISSUE THREAD

**Issue by lcofre** [apigee-127/swagger-test-templates#154]
It seems to me that when basePath is present in the swagger definition the requestData configuration is ignored. I tested the following reusing the files in the `swagger-test-templates/test/request-data/` test:

The swagger definition is the one at https://github.com/apigee-127/swagger-test-templates/blob/92f444daae2e99cb9a183c7cee85add74ff37522/test/request-data/swagger-post.json only adding the basePath property:

```
{
  "swagger": "2.0",
  "info": {
      "version": "0.0.0",
      "title": "Simple API"
  },
  "host": "api.uber.com",
  "basePath": "/api",
  "schemes": [
      "https"
  ],
  "paths": {
      "/user": {
          "post": {
              "parameters": [
                  {
                      "name": "latitude",
                      "in": "body",
                      "description": "Latitude component of location.",
                      "required": true,
                      "type": "number",
                      "format": "double"
                  },
                  {
                      "name": "longitude",
                      "in": "query",
                      "description": "longitude component of location.",
                      "required": true,
                      "type": "number",
                      "format": "double"
                  }
              ],
              "responses": {
                  "200": {
                      "description": "OK"
                  },
                  "400": {
                      "description": "NOT OK"
                  },
                  "500": {
                      "description": "SERVER ERROR"
                  }
              }
          }
      }
  },
  "definitions": {
      "User": {
          "type": "object",
          "properties": {
              "id": {
                  "type": "integer"
              },
              "username": {
                  "type": "string"
              }
          }
      }
  }
}
```

and the generation of the tests:
```
var fs = require("fs"),
stt = require("swagger-test-templates"),
swagger = require("./swagger.json"),
config = {
		assertionFormat: "expect",
		testModule: "request",
		pathName: [],
		maxLen: -1,
		requestData: {
			'/user': {
				post: {
					200: [{body: {"my-id": 2}, description: 'some description'}]
				}
			}
		}
};

var tests = stt.testGen(swagger, config);

tests.forEach(function(element) {
		fs.writeFile("./" + element.name, element.test, (err) => {
				if (err) throw err;

				console.log(element.name, " written");
		});
}, this);
```

Note that in the generated file user-test.js body contains
```
body: {
          latitude: 'DATA GOES HERE'
}
```

but if you remove the basePath property on the swagger definition you get

```
body: {"my-id":2}
```

Thanks very much for this cool project and for your readiness too!

**Comment by baynezy** [comment]
@lcofre - we had a similar bug reported the other day #150 
@noahdietz - has an open PR #153 which resolved #150 but we should test the case and resolve them both in one go.

Can you try the work around mentioned in #150 in the meantime?

**Comment by lcofre** [comment]
Great! I'll wait for the merge of the PR. In the meantime I am using the workaround.

Thanks!

**Comment by noahdietz** [comment]
@lcofre the fix for the base path bug in request data has been released in v1.4.3

Please follow up if there are any more issues. Thanks for the report and using STT!
